### PR TITLE
test: test that hits panic becasue of no column meta data

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Reproducer for #4695 

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: parquet_file/tests/metadata.rs [hunk 1]
```diff
@@ -103,6 +103,62 @@ async fn test_decoded_iox_metadata() {
     );
 }
 
+// Reproducer for "https://github.com/influxdata/influxdb_iox/issues/4695"
+// TODO: remove #[ignore] to tur the test on after the fix
```
Comment (NGA-TRAN) [inline parquet_file/tests/metadata.rs:107]:
```suggestion
// TODO: remove #[ignore] to turn the test on after the fix
```

File: parquet_file/tests/metadata.rs [hunk 2]
```diff
@@ -103,6 +103,62 @@ async fn test_decoded_iox_metadata() {
...
+    // Decode the IOx metadata embedded in the parquet file metadata.
+    let decoded = iox_parquet_meta
+        .decode()
+        .expect("failed to decode parquet file metadata");
+
+    let schema = decoded.read_schema().unwrap();
+    let col_summary = decoded
+        .read_statistics(&*schema)
+        .expect("Invalid Statistics");
+    assert!(col_summary.is_empty());
```
Comment (NGA-TRAN) [inline parquet_file/tests/metadata.rs:159]:
```suggestion
```

File: parquet_file/tests/metadata.rs [hunk 3]
```diff
@@ -103,6 +103,62 @@ async fn test_decoded_iox_metadata() {
...
+
+    // Decode the IOx metadata embedded in the parquet file metadata.
+    let decoded = iox_parquet_meta
+        .decode()
+        .expect("failed to decode parquet file metadata");
+
+    let schema = decoded.read_schema().unwrap();
+    let col_summary = decoded
+        .read_statistics(&*schema)
+        .expect("Invalid Statistics");
```
Comment (NGA-TRAN) [inline parquet_file/tests/metadata.rs:158]:
```suggestion
        .expect("Invalid Statistics");  // panic due to the bug
```

File: parquet_file/tests/metadata.rs [hunk 4]
```diff
@@ -103,6 +103,62 @@ async fn test_decoded_iox_metadata() {
...
+// TODO: remove #[ignore] to tur the test on after the fix
+#[ignore]
+#[tokio::test]
+async fn test_decoded_iox_metadata_without_data() {
+    // A representative IOx data sample (with a time column, an invariant upheld
+    // in the IOx write path)
+    let data = [
+        (
+            TIME_COLUMN_NAME,
+            to_timestamp_array(&[]), // No data on purpose
```
Comment (NGA-TRAN) [inline parquet_file/tests/metadata.rs:116]:
```suggestion
            to_timestamp_array(&[]), // No data on purpose to reproduce the panic bug
```

File: parquet_file/tests/metadata.rs [hunk 5]
```diff
@@ -103,6 +103,61 @@ async fn test_decoded_iox_metadata() {
...
+        sort_key: None,
+    };
+
+    let batch = RecordBatch::try_from_iter(data).unwrap();
+    let stream = futures::stream::iter([Ok(batch.clone())]);
+
+    let object_store: Arc<DynObjectStore> = Arc::new(object_store::memory::InMemory::default());
+    let storage = ParquetStorage::new(object_store);
+
+    let (iox_parquet_meta, _file_size) = storage
```
[the remark this answers is not in the document]
Comment (NGA-TRAN) [reply parquet_file/tests/metadata.rs:145]: Yes, the fix should be avoid writing empty data to parquet file as I just changed the title of the bug #4695 . This is just a reproducer

File: parquet_file/tests/metadata.rs [hunk 6]
```diff
@@ -103,6 +103,61 @@ async fn test_decoded_iox_metadata() {
...
+        .upload(stream, &meta)
+        .await
+        .expect("failed to serialise & persist record batch");
```
[the remark this answers is not in the document]
Comment (NGA-TRAN) [reply parquet_file/tests/metadata.rs:148]: Hah, Thisis a copy from @domodwyer 's test which he uses that word 😄

File: parquet_file/tests/metadata.rs [hunk 7]
```diff
@@ -103,6 +103,61 @@ async fn test_decoded_iox_metadata() {
...
+    // Decode the IOx metadata embedded in the parquet file metadata.
+    let decoded = iox_parquet_meta
+        .decode()
+        .expect("failed to decode parquet file metadata");
+
+    let schema = decoded.read_schema().unwrap();
+    let col_summary = decoded
+        .read_statistics(&*schema)
+        .expect("Invalid Statistics"); // panic due to the bug
+}
```
[the remark this answers is not in the document]
Comment (NGA-TRAN) [reply parquet_file/tests/metadata.rs:159]: Yes, that is for different bug and the test is here https://github.com/influxdata/influxdb_iox/blob/dde3c3922c321e6c4008eaf8b05bab72aebd503d/parquet_file/src/metadata.rs#L1026. When we fix the bug that return `not empty`, we will check the content too 😄

File: parquet_file/tests/metadata.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (alamb) [reply parquet_file/tests/metadata.rs:148]: yeah -- see https://github.com/influxdata/influxdb_iox/pull/4717 -- it is on my mind

**Pull request merged**